design-documents/graph-ql/coverage/returns.graphqls

```diff
@@ -0,0 +1,197 @@
+    return: Return
+}
+
+input RemoveReturnTrackingInput {
+    return_tracking_id: ID!
+}
+
+type RemoveReturnTrackingOutput {
+    return: Return
+}
+
+enum CustomAttributesPageEnum {
+    RETURN_ITEM_EDIT_FORM
+    RETURN_ITEMS_LISTING
+    # See https://github.com/magento/architecture/blob/master/design-documents/graph-ql/framework/attributes-metadata.md
+    # PRODUCTS_COMPARE
+    # PRODUCTS_LISTING
+    # ADVANCED_CATALOG_SEARCH
+}
+
+type Customer {
+    returns(
+        pageSize: Int = 20 @doc(description: "Specifies the maximum number of results to return at once. Defaults to 20."),
+        currentPage: Int = 1 @doc(description: "Specifies which page of results to return. The default value is 1."),
+    ): Returns @doc(description: "Information about the customer returns.")
+    return(id: ID!): Return @doc(description: "Get customer return details by its ID.")
+}
+
+type CustomerOrder {
+    returns(
+        pageSize: Int = 20 @doc(description: "Specifies the maximum number of results to return at once. Defaults to 20."),
+        currentPage: Int = 1 @doc(description: "Specifies which page of results to return. The default value is 1."),
+    ): Returns @doc(description: "Information about the order returns.")
+    items_eligible_for_return: [OrderItemInterface] @doc(description: "A list of order items eligible for return.")
+}
+
+type Returns {
+    items: [Return] @doc(description: "List of returns")
+    page_info: SearchResultPageInfo @doc(description: "Pagination metadata")
+    total_count: Int @doc(description: "Total count of customer returns")
+}
+
+type Return @doc(description: "Customer return") {
+    id: ID! @doc(description: "Return number")
+    order_id: String! @doc(description: "ID of the order based on which the return was created.")
+    creation_date: String! @doc(description: "The date when the return was requested.")
+    customer_email: String! @doc(description: "Email of the customer who created the return.")
+    customer_name: String @doc(description: "The name of the person returning the item")
+    status: ReturnStatus @doc(description: "Return status.")
+    shipping: ReturnShipping @doc(description: "Shipping information for the return.")
+    comments: [ReturnComment] @doc(description: "A list of comments posted for the return.")
+    items: [ReturnItem] @doc(description: "A list of items being returned.")
+    available_shipping_carriers: [ReturnShippingCarrier] @doc(description: "A list of shipping carriers available for returns.")
+}
+
+type ReturnItem {
+    id: ID!
+    product: ProductInterface!
+    custom_attributes: [CustomAttribute]
+    request_quantity: Float!
+    quantity: Float!
+    status: ReturnItemStatus!
+}
+
+# See https://github.com/magento/architecture/blob/master/design-documents/graph-ql/custom-attributes-container.md
+type CustomAttribute {
+    id: ID!
+    label: String!
+    value: String! @doc(description: "JSON encoded value of the attribute.")
```

Comment:
The linked document shows the `value` field here as just being a plain old string. Can you explain the comment about JSON-encoding?

Reply:
Look for `Here value is JSON-encoded value of the custom attribute.` in https://github.com/magento/architecture/blob/master/design-documents/graph-ql/custom-attributes-container.md
Maybe I do not understand the question.

Reply:
![image](https://user-images.githubusercontent.com/5233399/88237031-fac04080-cc43-11ea-8abe-ca46ecb55837.png)

Which part of this response would be in the JSON-encoded `value` field? Is it an actual JSON object?

Reply:
At least, `value` of multiselect attribute should be encoded somehow, there could be more examples of custom types of EAV attributes that need this.

Even though multiselect is not supported by RMA EAV attributes in the native Magento admin UI, I would prefer to keep the agreement that JSON is possible here. 

When the value is a string - it is still just a string in JSON too, so no complication here.